conversant/prompt_chatbot.py

```diff
@@ -186,11 +188,103 @@ def _update_max_context_examples(
+
+        Used for all responses besides the initial one. 
+
+        Args:
+            query (str): A query passed to the prompt chatbot.
+            
+            is_from_scratch (bool): Tells the chatbot if the reply
+            
+            should be generated from scratch or it is the continuation
+            
+            of the previous chunk
+            
+        Returns:
+            Interaction: Dictionary of query and generated LLM response
+            
+            bool: Indicates if the generation should be stopped
+        """
+        current_prompt = self.get_current_prompt(query)
+
+        current_prompt_size = self.co.tokenize(current_prompt).length
+
+        if current_prompt_size > self.max_prompt_size:
+            max_context_examples = self._update_max_context_examples(
+                current_prompt_size, self.chatbot_config["max_context_examples"]
+            )
+            current_prompt = self.get_current_prompt(query, max_context_examples)
+
+        elif (
+            self.curr_max_context_examples
+            != self.chatbot_config["max_context_examples"]
+        ):
+            warnings.warn(
+                "The max_context_examples value returned"
+                f" to {self.chatbot_config['max_context_examples']} - "
+                f"value set in the original config"
+            )
+      
+        generated_object = self.co.generate(
+            model=self.client_config["model"],
+            prompt=current_prompt,
+            max_tokens=TOKENS_PER_REQUEST,
+            temperature=self.client_config["temperature"],
+            frequency_penalty=self.client_config["frequency_penalty"],
+            presence_penalty=self.client_config["presence_penalty"],
+            stop_sequences=self.client_config["stop_sequences"],
+        )
+        # If response was cut off by .generate() finding a stop sequence,
+        # remove that sequence from the response.
+        response = generated_object.generations[0].text
+
+        is_final_chunk = False
+        if self.should_stop(response) or response=="":
+            is_final_chunk = True
+            for stop_seq in self.client_config["stop_sequences"]:
+                if response.endswith(stop_seq):
+                    response = response[: -len(stop_seq)]
+
+        if is_from_scratch:
+            response = response.lstrip()
+            self.chat_history.append(self.prompt.create_interaction(query, response))
+            self.prompt_size_history.append(
+                self.co.tokenize(
+                    self.prompt.create_interaction_string(query, response)
+                ).length
+            )
+            self.prompt_history.append(current_prompt)
+            
+        elif response != "": 
+            self.chat_history[-1]['bot'] += response
+            self.prompt_history[-1] += response 
+            self.prompt_size_history[-1] += self.co.tokenize(response).length
+          
```

Comment:
does it make sense to have a default else case here?

Reply:
I think it's fine because if it is neither, we do not want the chat history to change so we do not want to perform any actions🤔